Improve FLAC Pictures.

lubosz opened the pull request:
This patch exposes the picture functions for non-Ogg FLAC files, as specified in `flacfile.h`.

The type that pictureList uses was already conveniently exposed from the previous work on OGG+FLAC pictures.
I tested all 4 functions and they do work.

I was able to use only `DEF_SIMPLE_METHOD`.

The `XiphComment` exposes very similar picture functions differently, it does use a custom function `addPictureWithOwnership`, which I was not able to run due to a C++/Python type mismatch exception.

lubosz committed:
rest: Expose picture methods from FLAC:File.

Adds support for pictures in non-Ogg FLAC files as defined in flacfile.h.
Overload addPicture for auto_ptr<FLAC::Picture> and FLAC::Picture*.

Add pictures to flac test.

Test adding Picture from one FLAC file to another and remove functions.

Add la-with-cover-art.flat, update copyright file.

palfrey commented on src/wrapper/rest.cpp:
```diff
@@ -291,6 +291,10 @@ void exposeRest()
+      .DEF_SIMPLE_METHOD(pictureList)
+      .DEF_SIMPLE_METHOD(removePictures)
+      .DEF_SIMPLE_METHOD(removePicture)
+      .DEF_SIMPLE_METHOD(addPicture)
```
Can we have some Python code testing each of these please? Doesn't need to be complicated, provided it calls it without errors.

lubosz committed:
test/ogg: Show that ogg test fails.

Using the ByteVector constructor of FLAC::Picture expects a serialized
byte stream instead of only the picture bytes.

taglib fails silently here by printing "TagLib: Invalid picture block".

Check for non-zero bytes to check the validity of the picture, which
makes the test fail.

lubosz committed:
test: Add flac test.

Add a minimal test that copies tags from mp3 to flac.

lubosz committed:
rest: Use default constructor for FLAC::Picture.

Switch to default constructor in favor of ByteVector one, which assumes
serialized object bytes and not the picture bytes.

Expose setData, which actually takes pictures bytes.

test/ogg: Fix usage of FLAC::Picture.

lubosz committed:
rest: Use DEF_SIMPLE_METHOD consistently in FLAC::Picture.

The macro exactly does the same as def(#NAME, &cl::NAME).

lubosz committed:
rest: Expose more FLAC::Picture functions.

Add test for new FLAC::Picture functions.

lubosz commented:
So it turned out my previous patch was incomplete.

## Inappropriate `FLAC::Picture` constructor and more props

I noticed that the OGG test actually failed when initializing the `FLAC::Picture`.
The error message from taglib was:
`TagLib: Invalid picture block.`

This is due to the ByteVector constructor actually assuming a serialized byte stream of `FLAC::Picture` that can be retrieved with the `render()` method and not the picture bytes.

I have switched to the default constructor, fixing the ogg test.
I have added a test for flac.
I have exposed more properties for `FLAC:Picture` with a test.

## Exposing picture methods from `FLAC::File`

My previous naive exposing `addPicture` using `.DEF_SIMPLE_METHOD` was not correct.
I only tested it by using existing flac picture objects from a different file, which did work in terms of typing, but resolved in segfaults on exit.
It turns out that a function similar to `addPictureWithOwnership` from XiphComment is needed for smart pointers. This function is called for flac picture objects freshly created within Python.

In addition, another function that actually copies the data (by using the ByteVector constructor) is required for the pointer function. This function is needed for existing flac picture objects from other files.

It is possible to add an overloaded mapping, the correct one being chosen depending on the type.

I added tests for both cases.
